### PR TITLE
linsolve now able to solve unsimplified equations having const variable.

### LINKED ISSUE #10568: linsolve Returns EmptySet() if equation is not simplified.
```
>>> e = [2*x +y+3*z-1 , m*(x+3*y)+8*z-3 , n*(2*x +6*z) +8*y-5]
>>> linsolve(e, x,y,z)
{(-3/8, 5/8, 3/8)}
>>> e = [z-1 , m*(x+3*y)+8*z-3 , n*(2*x +6*z) +8*y-5]
>>> linsolve(e, x,y,z)
∅
>>> e = [z-1 , m*(x+3*y)+8*z-3 , y-5]
>>> linsolve(e, x,y,z)
∅
```

Answers are wrong.
In last  case, answer is simple : `{(-15 - 5/m, 5, 1)}`

also problem there is problem is this type : 

```
>>> e = [z-1 , m*(x+3*y)+8*z-3 , y-5]
>>> print linsolve(e, x,y)
EmptySet()
```

### PR DESCRIPTION
**Pull request by Shekharrajak**
Fixes https://github.com/sympy/sympy/issues/10568 One testcase 
Now it is able to solve 

```
>>> e = [z-1 , m*(x+3*y)+8*z-3 , y-5]
>>> linsolve(e, x,y,z)
`{(-15 - 5/m, 5, 1)
```

But still problem in this case 

```
>>> e = [z-1 , m*(x+3*y)+8*z-3 , y-5]
>>> print linsolve(e, x,y)
EmptySet()
```

### CODE REVIEW

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -1061,6 +1062,7 @@ def linear_eq_to_matrix(equations, *symbols):
     row_no = 1
 
     for equation in equations:
+        equation = ratsimp(equation)
         f = sympify(equation)
```
Comment (smichr) [inline sympy/solvers/solveset.py]: There is no need for `sympify` after `ratsimp` -- the return value will be sympified. so `f = ratsimp(equation)` would be fine (provided `equation` is not referrenced again; and if it is, then `f = equation = ratsimp(equation)` would suffice).

Comment (Shekharrajak) [inline sympy/solvers/solveset.py]: Yeah done. Thanks.

**Comment by Shekharrajak** [comment]
It's weird, `rsolve_hyper` giving wrong answer for this changes in this [testcase](https://travis-ci.org/sympy/sympy/jobs/108708868#L3666)

**Comment by Shekharrajak** [comment]
Ping @smichr @asmeurer @aktech

File: sympy/solvers/solveset.py [hunk 2]
```diff
@@ -894,7 +895,7 @@ def linear_eq_to_matrix(equations, *symbols):
     row_no = 1
 
     for equation in equations:
-        f = sympify(equation)
+        f = collect(equation.expand() ,symbols)
```
Comment (Shekharrajak) [inline sympy/solvers/solveset.py]: To collect coefficients of `symbols`.

**Comment by Shekharrajak** [comment]
Ping @hargup

**Comment by czgdp1807** [comment]
On current `master`,
**Code**
```python
from sympy import *
x, y, z, t = symbols('x y z t')
k, m, n = symbols('k m n', integer=True)
f, g, h = symbols('f g h', cls=Function)
init_printing()
def rep(eq, sols):
    print(eq, sols)
    for eq in e:
        sols = [sol for sol in s]
        print(simplify(eq.subs(((x, sols[0][0]), (y, sols[0][1]), (z, sols[0][2])))))
    print("========")
e = [2*x +y+3*z-1 , m*(x+3*y)+8*z-3 , n*(2*x +6*z) +8*y-5]
s = linsolve(e, x,y,z)
rep(e, s)
e = [z-1 , m*(x+3*y)+8*z-3 , n*(2*x +6*z) +8*y-5]
s = linsolve(e, x,y,z)
rep(e, s)
e = [z-1 , m*(x+3*y)+8*z-3 , y-5]
s = linsolve(e, x,y,z)
rep(e, s)
```
**Output**
```python
[2*x + y + 3*z - 1, m*(x + 3*y) + 8*z - 3, n*(2*x + 6*z) + 8*y - 5] FiniteSet((3*(6*m*n - 15*m - 6*n + 16)/(4*(3*m*n + 6*m + 4*n - 32)), -(6*m*n - 15*m - 34*n + 80)/(12*m*n + 24*m + 16*n - 128), -(5*m*(n - 5) + (m - 6)*(n - 8))/(15*m*n - (3*m - 16)*(n - 8))))
0
0
0
========
[z - 1, m*(x + 3*y) + 8*z - 3, n*(2*x + 6*z) + 8*y - 5] FiniteSet(((-9*m*n + 15*m/2 + 20)/(m*(3*n - 4)), (6*m*n - 5*m - 10*n)/(2*m*(3*n - 4)), 1))
0
0
0
========
[z - 1, m*(x + 3*y) + 8*z - 3, y - 5] FiniteSet((-15 - 5/m, 5, 1))
0
0
0
========
```
It looks like the issue has been fixed. Closing it for now. However if you feel that this should be re-opened, then please let us know in the comments below. Thanks for the contributions.

**Pull request closed without merging**